pyatlan/model/fields/atlan_fields.py

```diff
@@ -0,0 +1,528 @@
+from abc import ABC
+from typing import Union
+
+from pydantic import StrictBool, StrictFloat, StrictInt, StrictStr
+
+from pyatlan.model.enums import SortOrder
+from pyatlan.model.search import (
+    Exists,
+    Match,
+    Prefix,
+    Query,
+    Range,
+    SearchFieldType,
+    SortItem,
+    Term,
+    Terms,
+)
+from pyatlan.model.typedef import AttributeDef
+
+
+class AtlanField(ABC):
+    """
+    Base enumeration of all attributes that exist in Atlan, so you do not have to remember their
+    exact spelling or capitalization.
+    """
+
+    atlan_field_name: StrictStr
+
+
+class RelationField(AtlanField):
+    """
+    Represents any field used to capture a relationship in Atlan, which is not inherently
+    searchable.
+    """
+
+    def __init__(self, atlan: StrictStr):
```

Comment:
The parameter should be called atlan_field_name not atlan. Same for the sub-classes